lib/rspec/support/spec/encoding_helpers.rb

```diff
@@ -0,0 +1,30 @@
+module RSpec
+  module Support
+    module EncodingHelpers
+      module_function
+      if String.method_defined?(:encoding)
+
+        def expect_identical_string(str1, str2, expected_encoding=str1.encoding)
+          str1.encoding == expected_encoding &&
+            str1.each_byte.to_a == str2.each_byte.to_a
+        end
+
+      else
+
+        def expect_identical_string(str1, str2)
+          str1.split(//) == str2.split(//)
+        end
+      end
+    end
+  end
+end
+require 'rspec/expectations'
+# Special matchers for comparing encoded strings so that
+# we don't run any expectation failures through the Differ,
+# which also relies on EncodedString. Instead, confirm the
+# strings have the same encoding and same bytes.
+RSpec::Matchers.define :be_identical_string do |expected|
+  match do |actual|
+    RSpec::Support::EncodingHelpers.expect_identical_string(actual, expected)
```

Comment:
I know this isn't really right, and now it no longer checks the optional expected encoding, but I thought I'd stub it out for discussion. Also added a comment as to why it's there.  I admit, though, my words confuse characters, codepoints, and bytes.

 see the encoding check as a pre-req for the byte check. I'm not sure if it should be in the same expectation, a separate one, or chained.